code/datums/antagonists/antag_datum.dm

```diff
@@ -1,87 +1,79 @@
-//The Datum, Antagonist. Handles various antag things via a datum.
 /datum/antagonist
-	var/mob/living/owner //who's our owner and accordingly an antagonist
-	var/some_flufftext = "yer an antag larry"
-	var/prevented_antag_datum_type //the type of antag datum that this datum can't coexist with; should probably be a list
-	var/silent_update = FALSE //if we suppress messages during on_gain, apply_innate_effects, remove_innate_effects, and on_remove
+	var/name = "Antagonist"
+	var/antag_flag = null						//Flag that the antag(user's preferences)
+
+	var/datum/mind/owner						//Mind that owns this datum
+
+	var/text_on_gain = ""						//Text that will be shown to the mob when he gains the datum
+	var/text_on_lose = "" 						//Text that will be shown to the mob when he loses the datum
+	var/silent = FALSE							//Silent will prevent the gain/lose texts to show
+
+	var/give_special_equipment = TRUE			//If the mob will receive some sort of gear when he is given the datum
+	var/remove_clown_mut = TRUE					//If it will remove the clown mutation on datum gain
+	var/has_objectives = TRUE 					//Whether or not the objectives will be generated when this datum is created
+	var/number_of_possible_objectives			//Number of objectives this datum will have
+	var/list/current_objectives = list()		//List of objectives this datum will have
+	var/list/possible_objectives = list()		//List of possible objectives
+
+	var/can_coexist_with_others = TRUE			//Whether or not the person will be able to have more than one datum
+	var/list/typecache_datum_blacklist = list()	//List of datums this type can't coexist with
 
```

Comment:
Suggestion:  Perhaps have a variable for the ORIGINAL body?  I know some antags use this, such as traitor cyborgs.  (If you  debrain a traitor borg and shove em in a new borg body, they stop being an antag, IIRC)